src/Database/Redis/Connection.hs

```diff
@@ -266,18 +313,67 @@ shardMapFromClusterSlotsResponse ClusterSlotsResponse{..} = ShardMap <$> foldr m
             Cluster.Node clusterSlotsNodeID role hostname (toEnum clusterSlotsNodePort)
 
 refreshShardMap :: Cluster.Connection -> IO ShardMap
-refreshShardMap (Cluster.Connection nodeConns _ _ _ _) =
-    refreshShardMapWithNodeConn (head $ HM.elems nodeConns)
+refreshShardMap (Cluster.Connection nodeConnsVar _ shardMapVar _ _ Cluster.TcpInfo { idleTime, maxResources, timeoutOpt, connectAuth, connectTLSParams }) = do
+    putStrLn "Refresh called."
+    (oldShardMap, lastUpdated) <- atomically $ readTMVar shardMapVar
+    now <- Time.getCurrentTime
+    if Time.diffUTCTime now lastUpdated <= shardMapRefreshDelta
```

Comment:
When this will be the case? We can make MVar as empty to implement multiple threads not invoking the refreshShardMap at same time rather than maintain this complex logic of using timestamps.

Reply:
@aravindgopall problem here is: We want to make reads as lock free. So basically when a thread reads the MVar and till the time it gets an exception because of topology change and tries to update the MVar, other threads might also come and try to update this.
To avoid multiple threads updating the shardMap at the same time this is being done. There should be a cleaner way of doing this but I felt this is okay to unblock this commit merge ?

Reply:
Anyways, the timestamp check should only happen in case of exceptions because of topology change, so not much of a perf difference too.